### PR TITLE
Search resources mnrva 522

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-522

 ## Description:
Search Resources

 ## Testing:
1. Navigate to Resources list `/resources`
2. Begin typing, the results will update after entering 2 characters

 ## Screenshots:
searchResources

### CODE REVIEW

File: src/app/_features/resources/components/search/search.component.ts [hunk 1]
```diff
@@ -33,20 +35,27 @@ export class SearchComponent implements AfterViewInit, OnDestroy {
     .pipe(
       map((e:any) => e.target.value), // retrieve the value of the input
       filter((text:string) => text && text.length > 1), // filter if empty or more than 1
-      debounceTime(100), // search after 500 ms
+      debounceTime(100), // search after 100 ms
```
Comment (lovedeep92) [inline src/app/_features/resources/components/search/search.component.ts:38]: Could you make it 1000 in place of 100! I would work well and will reduce our api calls!

Comment (FarajiA) [reply src/app/_features/resources/components/search/search.component.ts:38]: Yup, this is updated.

**Comment by FarajiA** [comment]
@lovedeep92 yup all great points. 
I've updated this branch, any selected row is now included in the search whether it is apart of the search string or not.
searchResourcesII

**Comment by FarajiA** [comment]
So I think I see what you're saying:

1. After entering a string with 2 or more characters the search executes
2. When backspacing on this same string each deletion causes the search to execute

This seems correct as the backspacing of the text creates a different string and this will return a different result (though that may not be reflected in our scenario since we have so few resources). For the observable there is a distinction being made for _new_
 strings and won't make any duplicate requests of the same string within the 1 second interval.
https://github.com/racker/Minerva/pull/117/files#diff-89d9a1d4c9c0a3c4774d0402602f483bR48

If you think this should work differently lay out the flow and we can give it a shot.

**Comment by lovedeep92** [comment]
> So I think I see what you're saying:
> 
> 1. After entering a string with 2 or more characters the search executes
> 2. When backspacing on this same string each deletion causes the search to execute
> 
> This seems correct as the backspacing of the text creates a different string and this will return a different result (though that may not be reflected in our scenario since we have so few resources). For the observable there is a distinction being made for _new_
> strings and won't make any duplicate requests of the same string within the 1 second interval.
> https://github.com/racker/Minerva/pull/117/files#diff-89d9a1d4c9c0a3c4774d0402602f483bR48
> 
> If you think this should work differently lay out the flow and we can give it a shot.

@FarajiA , Yeah I am looking into it if we can make something. I think in place of keyup, can't we use onFocus and onBlur?

**Comment by FarajiA** [comment]
Updated debounce ms.

**Pull request merged**